go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -0,0 +1,484 @@
+				c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToGetOVNLoadBalancer",
+					"Error trying to obtain the OVN LoadBalancer for Service %s/%s: %v", name, namespace, err)
+				return err
+			}
+			// create the vip
+			vip := util.JoinHostPortInt32(ip, svcPort.Port)
+			klog.V(4).Infof("Updating service %s/%s with VIP %s %s", name, namespace, vip, svcPort.Protocol)
+			// get the endpoints associated to the vip
+			family := v1.IPv4Protocol
+			if utilnet.IsIPv6String(ip) {
+				family = v1.IPv6Protocol
+			}
+			eps := getLbEndpoints(endpointSlices, svcPort, family)
+			// Reconcile OVN
+			if err := loadbalancer.UpdateLoadBalancer(lbID, vip, eps); err != nil {
+				c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToUpdateOVNLoadBalancer",
+					"Error trying to update OVN LoadBalancer for Service %s/%s: %v", name, namespace, err)
+				return err
+			}
+			c.serviceTracker.updateService(name, namespace, vip, svcPort.Protocol)
+			// TODO: Insert a reject ACL if the service doesn't have endpoints associated
+			if len(eps) == 0 {
+				klog.V(4).Infof("Service %s/%s without endpoints", name, namespace)
+			}
+			// Node Port
+			if svcPort.NodePort != 0 {
+				gatewayRouters, _, err := gateway.GetOvnGateways()
+				if err != nil {
+					return errors.Wrapf(err, "failed to retrieve OVN gateway routers")
+				}
+				for _, gatewayRouter := range gatewayRouters {
+					lbID, err := gateway.GetGatewayLoadBalancer(gatewayRouter, svcPort.Protocol)
+					if err != nil {
+						klog.Warningf("Service Sync: Gateway router %s does not have load balancer (%v)",
+							gatewayRouter, err)
+						// TODO: why continue? should we error and requeue and retry?
+						continue
+					}
+					physicalIPs, err := gateway.GetGatewayPhysicalIPs(gatewayRouter)
+					if err != nil {
+						klog.Warningf("Service Sync: Gateway router %s does not have physical ips: %v",
+							gatewayRouter, err)
+						// TODO: why continue? should we error and requeue and retry?
+						continue
+					}
+					// With the physical_ip:port as the VIP, add an entry in 'load balancer'.
+					for _, physicalIP := range physicalIPs {
+						// only use the IPs of the same ClusterIP family
+						if utilnet.IsIPv6String(physicalIP) != utilnet.IsIPv6String(ip) {
+							continue
+						}
+						vip := util.JoinHostPortInt32(physicalIP, svcPort.NodePort)
+						klog.V(4).Infof("Updating NodePort service %s/%s with VIP %s %s", name, namespace, vip, svcPort.Protocol)
+						// Reconcile OVN
+						if err := loadbalancer.UpdateLoadBalancer(lbID, vip, eps); err != nil {
+							c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToUpdateOVNLoadBalancer",
+								"Error trying to update OVN LoadBalancer for Service %s/%s: %v", name, namespace, err)
+							return err
+						}
+						c.serviceTracker.updateService(name, namespace, vip, svcPort.Protocol)
+					}
+					// TODO: Insert a reject ACL if the service doesn't have endpoints associated
+					if len(eps) == 0 {
```

Comment:
This check is not needed, it's already performed on L274...unless `eps` have changed since then? But I didn't see that

Reply:
WIP, loadbalancer for clusterIP and noreport are different so their ACLs